lib/Cake/Test/Case/Controller/ApplicationControllerTest.php

```diff
@@ -0,0 +1,106 @@
+<?php
+App::uses('AppController', 'Controller');
+/**
+ * TransSessionIdController class for testing session.use_trans_sid=1.
+ *
+ * @package       Cake.Test.Case.Controller
+ */
+class TransSessionIdController extends AppController {
+
+/**
+ * Constructor.
+ *
+ * @param CakeRequest $request Request object for this controller.
+ * @param CakeResponse $response Response object for this controller.
+ */
+	public function __construct($request = null, $response = null) {
+		parent::__construct($request, $response);
+		$ini = Configure::read('Session.ini');
+		$ini['session.use_cookies'] = 0;
+		$ini['session.use_only_cookies'] = 0;
+		$ini['session.use_trans_sid'] = 1;
+		Configure::write('Session.ini', $ini);
+	}
+
+/**
+ * For testing redirect URL with session.use_trans_sid=1.
+ *
+ * @return CakeResponse|null
+ */
+	public function next() {
+		$sessionName = session_name();
+		$sessionId = $this->Session->id();
+		return $this->redirect(array(
+			'controller' => 'trans_session_id',
+			'action' => 'next_step',
+			'?' => array(
+				htmlspecialchars($sessionName) => htmlspecialchars($sessionId),
```

Comment:
Why are you HTML escaping URL parameters? The Location header isn't vulnerable to XSS. CakePHP will handle urlencoding this data.

Reply:
Right, that's not important in this case.